### ISSUE TITLE
On destroy ANR issue

### ISSUE THREAD

**Issue by rezafaraji93** [crisp-im/crisp-sdk-android#95]
I am using crisp 1.0.9 and exoplayer 2.17.0. When ever  we close the chat activity and on destroy ANR happens with this error:

```
java.lang.NoClassDefFoundError: Failed resolution of: Lcom/google/android/exoplayer2/Player$EventListener;
        at im.crisp.client.internal.utils.b.a(Unknown Source)
        at im.crisp.client.Crisp.d(Unknown Source)
        at im.crisp.client.ChatActivity.onDestroy(Unknown Source)
        at android.app.Activity.performDestroy(Activity.java:7255)
        at android.app.Instrumentation.callActivityOnDestroy(Instrumentation.java:1161)
        at android.app.ActivityThread.performDestroyActivity(ActivityThread.java:4657)
        at android.app.ActivityThread.handleDestroyActivity(ActivityThread.java:4697)
        at android.app.ActivityThread.-wrap7(ActivityThread.java)
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1720)
        at android.os.Handler.dispatchMessage(Handler.java:102)
        at android.os.Looper.loop(Looper.java:154)
        at android.app.ActivityThread.main(ActivityThread.java:6816)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1563)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1451)
     Caused by: java.lang.ClassNotFoundException: Didn't find class "com.google.android.exoplayer2.Player$EventListener" on path: DexPathList[[zip file "/data/app/com.faraji.raybit-2/base.apk"],nativeLibraryDirectories=[/data/app/com.faraji.raybit-2/lib/arm, /data/app/com.faraji.raybit-2/base.apk!/lib/armeabi-v7a, /system/lib, /vendor/lib]]
        at dalvik.system.BaseDexClassLoader.findClass(BaseDexClassLoader.java:56)
        at java.lang.ClassLoader.loadClass(ClassLoader.java:380)
        at java.lang.ClassLoader.loadClass(ClassLoader.java:312)
        at im.crisp.client.internal.utils.b.a(Unknown Source) 
        at im.crisp.client.Crisp.d(Unknown Source) 
        at im.crisp.client.ChatActivity.onDestroy(Unknown Source) 
        at android.app.Activity.performDestroy(Activity.java:7255) 
        at android.app.Instrumentation.callActivityOnDestroy(Instrumentation.java:1161) 
        at android.app.ActivityThread.performDestroyActivity(ActivityThread.java:4657) 
        at android.app.ActivityThread.handleDestroyActivity(ActivityThread.java:4697) 
        at android.app.ActivityThread.-wrap7(ActivityThread.java) 
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1720) 
        at android.os.Handler.dispatchMessage(Handler.java:102) 
        at android.os.Looper.loop(Looper.java:154) 
        at android.app.ActivityThread.main(ActivityThread.java:6816) 
        at java.lang.reflect.Method.invoke(Native Method) 
        at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1563) 
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1451) 
```

**Comment by rezafaraji93** [comment]
More log about this crash:

`Rejecting re-init on previously-failed class java.lang.Class<im.crisp.client.internal.utils.b$1>: java.lang.NoClassDefFoundError: Failed resolution of: Lcom/google/android/exoplayer2/Player$EventListener;
    at im.crisp.client.internal.utils.b im.crisp.client.internal.utils.b.a() ((null):-1)
    at void im.crisp.client.Crisp.d() ((null):-1)
    at void im.crisp.client.ChatActivity.onDestroy() ((null):-1)
    at void android.app.Activity.performDestroy() (Activity.java:7255)
    at void android.app.Instrumentation.callActivityOnDestroy(android.app.Activity) (Instrumentation.java:1161)
    at android.app.ActivityThread$ActivityClientRecord android.app.ActivityThread.performDestroyActivity(android.os.IBinder, boolean, int, boolean) (ActivityThread.java:4657)
    at void android.app.ActivityThread.handleDestroyActivity(android.os.IBinder, boolean, int, boolean) (ActivityThread.java:4697)
    at void android.app.ActivityThread.-wrap7(android.app.ActivityThread, android.os.IBinder, boolean, int, boolean) (ActivityThread.java:-1)
    at void android.app.ActivityThread$H.handleMessage(android.os.Message) (ActivityThread.java:1720)
    at void android.os.Handler.dispatchMessage(android.os.Message) (Handler.java:102)
    at void android.os.Looper.loop() (Looper.java:154)
    at void android.app.ActivityThread.main(java.lang.String[]) (ActivityThread.java:6816)
    at java.lang.Object java.lang.reflect.Method.invoke!(java.lang.Object, java.lang.Object[]) (Method.java:-2)
    at void com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run() (ZygoteInit.java:1563)
    at void com.android.internal.os.ZygoteInit.main(java.lang.String[]) (ZygoteInit.java:1451)
Caused by: java.lang.ClassNotFoundException: Didn't find class "com.google.android.exoplayer2.Player$EventListener" on path: DexPathList[[zip file "/data/app/com.faraji.raybit-2/base.apk"],nativeLibraryDirectories=[/data/app/com.faraji.raybit-2/lib/arm, /data/app/com.faraji.raybit-2/base.apk!/lib/armeabi-v7a, /system/lib, /vendor/lib]]
    at java.lang.Class dalvik.system.BaseDexClassLoader.findClass(java.lang.String) (BaseDexClassLoader.java:56)
    at java.lang.Class java.lang.ClassLoader.loadClass(java.lang.String, boolean) (ClassLoader.java:380)
    at java.lang.Class java.lang.ClassLoader.loadClass(java.lang.String) (ClassLoader.java:312)
    at im.crisp.client.internal.utils.b im.crisp.client.internal.utils.b.a() ((null):-1)
    at void im.crisp.client.Crisp.d() ((null):-1)
    at void im.crisp.client.ChatActivity.onDestroy() ((null):-1)
    at void android.app.Activity.performDestroy() (Activity.java:7255)
    at void android.app.Instrumentation.callActivityOnDestroy(android.app.Activity) (Instrumentation.java:1161)
    at android.app.ActivityThread$ActivityClientRecord android.app.ActivityThread.performDestroyActivity(android.os.IBinder, boolean, int, boolean) (ActivityThread.java:4657)
    at void android.app.ActivityThread.handleDestroyActivity(android.os.IBinder, boolean, int, boolean) (ActivityThread.java:4697)
    at void android.app.ActivityThread.-wrap7(android.app.ActivityThread, android.os.IBinder, boolean, int, boolean) (ActivityThread.java:-1)
    at void android.app.ActivityThread$H.handleMessage(android.os.Message) (ActivityThread.java:1720)
    at void android.os.Handler.dispatchMessage(android.os.Message) (Handler.java:102)
    at void android.os.Looper.loop() (Looper.java:154)
    at void android.app.ActivityThread.main(java.lang.String[]) (ActivityThread.java:6816)
    at java.lang.Object java.lang.reflect.Method.invoke!(java.lang.Object, java.lang.Object[]) (Method.java:-2)
    at void com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run() (ZygoteInit.java:1563)
    at void com.android.internal.os.ZygoteInit.main(java.lang.String[]) (ZygoteInit.java:1451)`

**Comment by rezafaraji93** [comment]
Any updates??

**Comment by eliottvincent** [comment]
Hey @rezafaraji93 , our Android developer is busy working on the rewrite of the new Android app. He will have a look at your issue when possible.

**Comment by rezafaraji93** [comment]
Hi, would you please tell me your timing to fix this issue? I need to know because everything is ready for production but crisp.

**Comment by Doc1faux** [comment]
I @rezafaraji93 and thank your for your feedback.
Sorry for the late in my reply, as @eliottvincent mentioned, I'm currently working on the brand new Android app.

The crash you met is due to a mismatch between versions of ExoPlayer the SDK uses and you import in your app.
Indeed, the SDK uses version `2.13.0` and you import the version `2.17.0`.
Google seems to have renamed the `com.google.android.exoplayer2.Player.EventListener` to `com.google.android.exoplayer2.Player.Listener` which is causing the crash.
I've tested the upgrade this morning and all seems to work as before on API 21+ (Android 5.0+) but I'm waiting for an issue from [Socket.IO](https://github.com/socketio/socket.io-client-java/issues/705) to be resolved to consider a new release.

In the meanwhile, maybe you could release your app with Google ExoPlayer `2.13.0`?
I keep you informed when a release on our side can be pushed.

**Comment by Doc1faux** [comment]
Fixed in `1.0.10` and released :)
